app/models/concerns/host_common.rb

```diff
@@ -38,15 +38,15 @@ module HostCommon
     def lookup_values_attributes=(lookup_values_attributes)
       lookup_values_attributes.each_value do |attribute|
         attr = attribute.dup
-        if attr.has_key? :id
-          lookup_value = lookup_values.find attr.delete(:id)
+        if id = attr.delete(:id)
+          lookup_value = self.lookup_values.to_a.select{|i| i.id == id}.first
```

Comment:
we find it like this so the object remains in the lookup_values array, this way, if it's marked for destruction, it will get destroyed when save is invoked.